go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -715,6 +715,34 @@ func (ddb *DoltDB) HasBranch(ctx context.Context, branchName string) (string, bo
 	return "", false, nil
 }
 
+// HasRemoteTrackingBranch returns whether the DB has a remote tracking branch with the name given, case-insensitive.
+// Returns the case-sensitive matching branch if found, as well as a bool indicating if there was a case-insensitive match,
+// remote tracking branchRef that is the only match for the branchName and any error.
+func (ddb *DoltDB) HasRemoteTrackingBranch(ctx context.Context, branchName string) (string, bool, ref.RemoteRef, error) {
+	remoteRefFound := false
+	var remoteRef ref.RemoteRef
+
+	remoteRefs, err := ddb.GetRemoteRefs(ctx)
+	if err == nil {
```

Comment:
Would be better to check if `err != nil` and return that `err` up, instead of swallowing it here.